MAINTAINERS: add orphaned RISC-V boards back to the RISC-V area

fkokosinski opened the pull request:
This PR restores the pre-HWMv2 status of RISC-V boards being covered by the RISC-V area of maintenance, by adding the following orphaned boards back to the area:
* `litex_vexriscv`
* `opentitan_earlgrey`
* `m2gl025_miv`
* `mpfs_icicle`

An example of an affected PR can be found here: https://github.com/zephyrproject-rtos/zephyr/pull/72569.

fkokosinski committed:
MAINTAINERS: add orphaned RISC-V boards back to the RISC-V area

This commit restores the pre-HWMv2 status of RISC-V boards being covered by
the RISC-V area of maintenance.

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

nashif commented on MAINTAINERS.yml:
```diff
@@ -2903,6 +2903,10 @@ RISCV arch:
+    - boards/enjoydigital/litex_vexriscv/
+    - boards/lowrisc/opentitan_earlgrey/
+    - boards/microchip/m2gl025_miv/
+    - boards/microchip/mpfs_icicle/
```
do not see why those would fall under the riscv architecture, they deserver an area maintained by microchip in this case, including the socs. Architecture are should not become a collection for boards/socs of that architecture.

fkokosinski replied:
Thanks for the comment. I've separated Microchip RISC-V platforms into a separate area. Please let me know if this works for you.

fkokosinski committed:
MAINTAINERS: introduce the `Microchip RISC-V Platforms` area

This commit introduces the `Microchip RISC-V Platforms` area. This area
coveres the following current Zephyr targets:
* `m2gl025_miv`
* `mpfs_icicle`

Signed-off-by: Filip Kokosinski <fkokosinski@antmicro.com>

kartben commented:
@nashif please confirm your comment has been addressed
